lib/vector-core/src/event/log_event.rs

```diff
@@ -261,6 +261,17 @@ impl From<String> for LogEvent {
     }
 }
 
+impl From<Value> for LogEvent {
```

Comment:
I don't think this is necessary, it's a little too implicit, and it doesn't seem right that `Value::Map` would get nested under `message_key`.